src/main/java/core/basesyntax/dao/animal/AnimalDaoImpl.java

```diff
@@ -3,20 +3,44 @@
 import core.basesyntax.dao.AbstractDao;
 import core.basesyntax.model.zoo.Animal;
 import java.util.List;
+import java.util.stream.Collectors;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
 public class AnimalDaoImpl extends AbstractDao implements AnimalDao {
     public AnimalDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
-    @Override
     public Animal save(Animal animal) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(animal);
+            transaction.commit();
+            return animal;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert animal " + animal, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Animal> findByNameFirstLetter(Character character) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Animal", Animal.class).getResultStream()
```

Comment:
let's use `like` operator here. https://thorben-janssen.com/jpa-2-2s-new-stream-method-and-how-you-should-not-use-it/